ietfYangDraftPull/pull_local.py

```diff
@@ -107,63 +110,32 @@ def main(script_conf: ScriptConfig = DEFAULT_SCRIPT_CONFIG.copy()) -> list[JobLo
     config_path = args.config_path
     config = create_config(config_path)
     notify_indexing = config.get('General-Section', 'notify-index')
-    config_name = config.get('General-Section', 'repo-config-name')
-    config_email = config.get('General-Section', 'repo-config-email')
     log_directory = config.get('Directory-Section', 'logs')
-    ietf_rfc_url = config.get('Web-Section', 'ietf-RFC-tar-private-url')
-    temp_dir = config.get('Directory-Section', 'temp')
+    yang_models_dir = config.get('Directory-Section', 'yang-models-dir')
+    non_ietf_directory = config.get('Directory-Section', 'non-ietf-directory')
+    openconfig_dir = os.path.join(non_ietf_directory, 'openconfig/public')
     logger = log.get_logger('pull_local', f'{log_directory}/jobs/pull-local.log')
     logger.info('Starting cron job IETF pull request local')
 
     messages = []
     notify_indexing = notify_indexing == 'True'
     success = True
-    try:
-        # Clone YangModels/yang repository
-        clone_dir = os.path.join(temp_dir, 'pull_local')
-        if os.path.exists(clone_dir):
-            shutil.rmtree(clone_dir)
-        repo = repoutil.ModifiableRepoUtil(
-            f'{github_url}/YangModels/yang.git',
-            clone_options=repoutil.RepoUtil.CloneOptions(
-                config_username=config_name,
-                config_user_email=config_email,
-                local_dir=clone_dir,
-            ),
-        )
-        logger.info(f'YangModels/yang repo cloned to local directory {repo.local_dir}')
-
-        response = requests.get(ietf_rfc_url)
-        tgz_path = os.path.join(repo.local_dir, 'rfc.tgz')
-        extract_to = os.path.join(repo.local_dir, 'standard/ietf/RFC')
-        with open(tgz_path, 'wb') as zfile:
-            zfile.write(response.content)
-        tar_opened = draftPullUtility.extract_rfc_tgz(tgz_path, extract_to, logger)
-
-        if tar_opened:
-            # Standard RFC modules
-            directory_success, message = populate_directory(extract_to, notify_indexing, logger)
-            success = success and directory_success
-            messages.append({'label': 'Standard RFC modules', 'message': message})
-
-        # Experimental modules
-        experimental_path = os.path.join(repo.local_dir, 'experimental/ietf-extracted-YANG-modules')
-
-        directory_success, message = populate_directory(experimental_path, notify_indexing, logger)
-        success = success and directory_success
-        messages.append({'label': 'Experimental modules', 'message': message})
-
-        # IANA modules
-        iana_path = os.path.join(repo.local_dir, 'standard/iana')
-
-        if os.path.exists(iana_path):
-            directory_success, message = populate_directory(iana_path, notify_indexing, logger)
```

Comment:
what about all this specific logic, isn't it needed anymore?

Reply:
As far as I can tell, the logic was separate only because we wanted to clone the openconfig repo every time. We no longer need to do that if we keep the openconfig repo in `/var/yang/nonietf` up to date, so there is no need for extra logic/

Reply:
not sure, for example, this part:
```
 response = requests.get(ietf_rfc_url)
        tgz_path = os.path.join(repo.local_dir, 'rfc.tgz')
        extract_to = os.path.join(repo.local_dir, 'standard/ietf/RFC')
        with open(tgz_path, 'wb') as zfile:
            zfile.write(response.content)
        tar_opened = draftPullUtility.extract_rfc_tgz(tgz_path, extract_to, logger)
```
makes sense because not all modules can be already available in the ```standard/ietf/RFC``` dir (the PR from us isn't automatically merged yet, for example), so we make sure that we populate only the latest modules using the ```ietf_rfc_url```

Reply:
it depends on when the script runs, for now, the ```ietf_push.py``` script runs at 5 14 and ```pull_local.py``` runs at 5 22, it's not enough time to finish the workflows in the ```yang-catalog``` and in the ```YangModels``` repos

Reply:
I see. Looking at past runs of the workflows, it looks like they both generally finish within 1 hour and 30 min. If we move `pull_local.py` to run 2 hours after `ietf_push.py` we should be fine, right?

Reply:
right, +-4 hours (~1.5h for fork + ~1.5h for YangModels + some delta), but I wouldn't consider this method a safe one for the future, to be honest, maybe it needs at least mentioned somewhere, if workflows take more time we will populate new/changed modules the next day, not sure if it's critical. Maybe we can use the strategy pattern here in some way and for specific worktrees use specific population strategies, which will in this example just update the modules in the right way, so our code will be more scalable for the future as well

Reply:
The fork actually only triggers the check for SDO modules, which runs in less than 10 mins https://github.com/yang-catalog/yang/actions. I just thought of something though, we could just run the populate script over the `/var/yang/ietf/YANG-rfc` directory if we don't want to wait for the repo.

Reply:
Yeah, basically, if that dir includes all the needed modules (not sure about this) then there’s no need to use the repo at all

Reply:
I believe the directories are both supposed to have the same contents, with the addition of the symlinks without revisions in standard/ietf/RFC. However it seem there are some differences
```
1,3d0
< example-dhcpv6-class-select@2022-06-20.yang
< example-dhcpv6-opt-sip-serv@2022-06-20.yang
< example-dhcpv6-server-conf@2022-06-20.yang
10a8
> iana-if-type@2017-01-19.yang
12a11
> iana-routing-types@2018-10-29.yang
31a31
> ietf-connectionless-oam-methods@2019-04-16.yang
132a133
> ietf-restconf-monitoring@2017-01-26.yang
134c135
< ietf-rip@2020-02-20.yang
---
> ietf-restconf@2017-01-26.yang
184a186
> ietf-yang-structure-ext@2020-06-17.yang
```
I think this is unintentional so I'll try and figure out what's causing this.

Reply:
So the modules which are present in `/var/yang/ietf/YANG-rfc` but not in `standard/ietf/RFC` are blacklisted in the `/var/yang/ietf-exceptions/exceptions.dat` file to prevent them from being pushed to the repo. This would still not happen in the pull script, so it's not an issue.

These modules
```
iana-if-type@2017-01-19.yang
iana-routing-types@2018-10-29.yang
ietf-connectionless-oam-methods@2019-04-16.yang
```
seem to have been added to `standard/ietf/RFC` manually by accident.

And the last three
```
ietf-restconf-monitoring@2017-01-26.yang
ietf-restconf@2017-01-26.yang
ietf-yang-structure-ext@2020-06-17.yang
```
aren't getting extracted anymore because of a xym error.
```
Extracting 'ietf-restconf'
   ERROR: '/var/yang/ietf/rfc/rfc8040.txt', Line 4761 - 'module' statement within another module
```